tests/cases/push-generator.js

```diff
@@ -674,15 +677,28 @@ PushGenerator.prototype.displayMessage = function() {
   this.createRequest().then(function(request) {
     var content = document.getElementById('message-info-dialog').cloneNode(true /* deep */);
     var headers = [];
+    var curlCommand = 'curl "' + request.url + '" --request POST';
 
     Object.keys(request.headers).forEach(function(headerName) {
       headers.push(headerName + ': ' + request.headers[headerName]);
+      curlCommand += ' --header "' + headerName + ': ' + request.headers[headerName] + '"';
     });
 
     content.querySelector('#endpoint').textContent = request.url;
     content.querySelector('#headers').textContent = headers.join('\n');
     content.querySelector('#body').textContent = request.body;
 
+    if (request.body instanceof ArrayBuffer) {
+      var curlError = 'Since web push protocol requires a stream as ' +
+          'the body of the request, there is no CURL command that will ' +
+          'stream an encrypted payload.';
+      content.querySelector('#error').textContent = curlError;
```

Comment:
It seems to me that either #error or #request would now always be presented as "[undefined]", which is a bit unfortunate. How about removing #error and hiding #request by default (through CSS `display: none`), and making it visible when we can populate the CURL request in the case below?

(I'm happy to do this in a follow-up :))